### PR TITLE
many: enumerate system seeds, return them on the /v2/systems API endpoint

### PR DESCRIPTION
**Pull request by bboozzoo**
Implement listing of system seeds and return the list on the /v2/systems snapd API endpoint. 

This is part of the recovery chooser work.

### CODE REVIEW

File: overlord/devicestate/devicemgr.go [hunk 1]
```diff
@@ -723,12 +726,75 @@ func (m *DeviceManager) Serial() (*asserts.Serial, error) {
...
-	// TODO:UC20 list available systems in the seed, load each with
-	// seed.LoadAssertions()
-	// TODO:UC20 convert brand-id to user friendly brand name
-	return nil, fmt.Errorf("not implemented")
+func (m *DeviceManager) Systems() ([]System, error) {
+	systemLabels, err := filepath.Glob(filepath.Join(dirs.SnapSeedDir, "systems", "*"))
+	if err != nil {
+		if os.IsNotExist(err) {
+			// maybe not a UC20 system
+			return nil, nil
```
Comment (pedronis) [inline overlord/devicestate/devicemgr.go]: maybe this should be an explicit ErrNoSystems ?

File: overlord/devicestate/devicemgr.go [hunk 2]
```diff
@@ -723,12 +726,75 @@ func (m *DeviceManager) Serial() (*asserts.Serial, error) {
...
+		label := filepath.Base(fpLabel)
+		model, brand, err := systemSeedModelAndBrand(label)
+		if err != nil {
+			// skip broken seed?
+			logger.Noticef("cannot load system %q seed: %v", label, err)
+			continue
+		}
+		systems = append(systems, System{
+			// TODO:UC20 check if current installation was done with that
+			// system
```
Comment (pedronis) [inline overlord/devicestate/devicemgr.go]: thanks for the explicit TODOs

File: overlord/devicestate/devicemgr.go [hunk 3]
```diff
@@ -723,12 +726,75 @@ func (m *DeviceManager) Serial() (*asserts.Serial, error) {
...
+			return nil, nil
+		}
+		return nil, fmt.Errorf("cannot list available systems: %v", err)
+	}
+	var systems []System
+	for _, fpLabel := range systemLabels {
+		label := filepath.Base(fpLabel)
+		model, brand, err := systemSeedModelAndBrand(label)
+		if err != nil {
+			// skip broken seed?
```
Comment (pedronis) [inline overlord/devicestate/devicemgr.go]: we might have a Broken string in the result like for snaps (see snap.Info.Broken?) ? maybe leave a TODO about that

File: daemon/api_systems.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+type systemsResponse struct {
+	Systems []client.System `json:"systems,omitempty"`
+}
+
+func getSystems(c *Command, r *http.Request, user *auth.UserState) Response {
+	var rsp systemsResponse
+
+	seedSystems, err := c.d.overlord.DeviceManager().Systems()
```
Comment (mvo5) [inline daemon/api_systems.go]: Nice

File: overlord/devicestate/devicemgr.go [hunk 5]
```diff
@@ -723,12 +727,80 @@ func (m *DeviceManager) Serial() (*asserts.Serial, error) {
...
+		return nil, nil, fmt.Errorf("cannot obtain model: %v", err)
+	}
+	brand, err = s.Brand()
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot obtain brand: %v", err)
+	}
+	return model, brand, nil
+}
+
+var ErrNoSystems = errors.New("no systems seeds")
```
Comment (pedronis) [inline overlord/devicestate/devicemgr.go]: let's change this to "no seed systems" in a follow up

**Pull request merged**